pkg/logs/pipeline/pipeline.go

```diff
@@ -57,72 +51,68 @@ func NewPipeline(outputChan chan *message.Message, processingRules []*config.Pro
 	}
 
 	inputChan := make(chan *message.Message, config.ChanSize)
-	processor := processor.New(inputChan, senderChan, processingRules, encoder, diagnosticMessageReceiver)
+	processor := processor.New(inputChan, strategyInput, processingRules, encoder, diagnosticMessageReceiver)
 
 	return &Pipeline{
 		InputChan: inputChan,
 		processor: processor,
-		sender:    logSender,
+		strategy:  strategy,
+		sender:    logsSender,
 	}
 }
 
 // Start launches the pipeline
 func (p *Pipeline) Start() {
 	p.sender.Start()
+	p.strategy.Start()
 	p.processor.Start()
 }
 
 // Stop stops the pipeline
 func (p *Pipeline) Stop() {
 	p.processor.Stop()
+	p.strategy.Stop()
 	p.sender.Stop()
 }
 
 // Flush flushes synchronously the processor and sender managed by this pipeline.
 func (p *Pipeline) Flush(ctx context.Context) {
 	p.processor.Flush(ctx) // flush messages in the processor into the sender
-	p.sender.Flush(ctx)    // flush the sender
 }
 
-func getMainDestinations(endpoints *config.Endpoints, destinationsContext *client.DestinationsContext) *client.Destinations {
+func getDestinations(endpoints *config.Endpoints, destinationsContext *client.DestinationsContext, pipelineID int) *client.Destinations {
+	reliable := []client.Destination{}
+	additionals := []client.Destination{}
+
 	if endpoints.UseHTTP {
-		main := http.NewDestination(endpoints.Main, http.JSONContentType, destinationsContext, endpoints.BatchMaxConcurrentSend)
-		additionals := []client.Destination{}
-		for _, endpoint := range endpoints.GetUnReliableAdditionals() {
-			additionals = append(additionals, http.NewDestination(endpoint, http.JSONContentType, destinationsContext, endpoints.BatchMaxConcurrentSend))
+		for i, endpoint := range endpoints.GetReliableEndpoints() {
+			telemetryName := fmt.Sprintf("logs_%d_reliable_%d", pipelineID, i)
+			reliable = append(reliable, http.NewDestination(endpoint, http.JSONContentType, destinationsContext, endpoints.BatchMaxConcurrentSend, true, telemetryName))
+		}
+		for i, endpoint := range endpoints.GetUnReliableEndpoints() {
+			telemetryName := fmt.Sprintf("logs_%d_unreliable_%d", pipelineID, i)
```

Comment:
non-blocking nit: I think the target expvar name generation should probably be the responsibility of the invoked method and not the caller. I think I saw another file with the same pattern too.

Reply:
The reason this is done here is to differentiate `logs` vs `epforwared` which is tagged differently. 
I was just trying to avoid adding 3 extra arguments to the already very long method signature that are only used for telemetry. WDYT?